### PR TITLE
cmd/tailscaled: disable netns in userspace-networking mode

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #2827
Updates #2822

### CODE REVIEW

File: net/netns/netns.go [hunk 1]
```diff
@@ -19,12 +19,23 @@ import (
 	"net"
 
 	"inet.af/netaddr"
+	"tailscale.com/syncs"
 )
 
+var disabled syncs.AtomicBool
+
+// Disable disables netns for the process.
+func Disable() {
```
Comment (DentonGentry) [inline net/netns/netns.go:28]: Recommend we log that netns is being disabled.

**Review by DentonGentry — APPROVED** [review]

File: net/netns/netns.go (hunk 1 above)
Comment (bradfitz) [reply net/netns/netns.go:28]: We already log that userspace-networking is being used, though? They're 1:1 so it doesn't strictly add any information.

I don't really mind but I'm trying to be a little bit more cautious of adding lots of logs.

Comment (DentonGentry) [reply net/netns/netns.go:28]: +1

Comment (bradfitz) [reply net/netns/netns.go:28]: yeah, right before this is:

```
                logf("wgengine.NewUserspaceEngine(tun %q) ...", name)
```

Where the `name` is `"userspace-networking"`

Comment (bradfitz) [reply net/netns/netns.go:28]: Or did you want to backport this to 1.14.x?

Comment (DentonGentry) [reply net/netns/netns.go:28]: I wasn't going to recommend that, the people running into it so far would likely be fine trying an unstable release.

**Pull request merged**